distributed/shuffle/_scheduler_plugin.py

```diff
@@ -383,8 +383,28 @@ def transition(
         **kwargs: Any,
     ) -> None:
         """Clean up scheduler and worker state once a shuffle becomes inactive."""
-        if finish not in ("released", "forgotten"):
+        if finish not in ("released", "erred", "forgotten"):
             return
+
+        if finish == "erred":
+            ts = self.scheduler.tasks[key]
+            for active_shuffle in self.active_shuffles.values():
+                barrier = self.scheduler.tasks[barrier_key(active_shuffle.id)]
+                if (
+                    ts == barrier
+                    or ts in barrier.dependents
+                    or ts in barrier.dependencies
+                ):
+                    self.scheduler.log_event(
+                        "p2p",
+                        {
+                            "action": "p2p-failed",
+                            "shuffle": active_shuffle.id,
+                            "stimulus": stimulus_id,
+                        },
+                    )
```

Comment:
We'd receive 2N + 1 event logs where N is the number of partitions if a shuffle fails, wouldn't we?

Reply:
Yes (in the worst case). We can reduce this through caching/early invalidation if necessary.

Reply:
I think this is worth it. If we bother to put in a non trivial transition hook like this we can also cache this. It will also make the event much more useful since every time we'd use this we'd have to deduplicate it anyhow

Reply:
Done